### PR TITLE
Interfaces and conditional routing

### PR DESCRIPTION
**Pull request by bobbiejc**
Added example showing how to use interfaces, @supplies and conditional routing

### CODE REVIEW

File: interfaces/rerank/README.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
+`@supplies` directive connects a field selection on a concrete type to an interface request.
```
Comment (ddebrunner) [inline interfaces/rerank/README.md:1]: No it doesn't :-)

File: interfaces/rerank/README.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+interface JSONDocs {
+    docs: JSON
+}
+
+type Concrete implements JSONDocs {
+    docs: JSON
+}
+```
+
+This snippet demonstrates the alternative execution paths, with `reRank` field selection being *supplied* by the fields `reRankAsc`, `reRankDesc`, and `reRankNoop`.
```
Comment (ddebrunner) [inline interfaces/rerank/README.md:19]: Only issue is that the reader is left on their own to figure out how any of this is done. Some more explanation in the comments would have helped, e.g. point the reader towards the 'if' argument in the supplies and how it is used.

File: interfaces/rerank/README.md (hunk 1 above)
Comment (bobbiejc) [reply interfaces/rerank/README.md:1]: ha -- I thought I copied that from the documentation!

**Review by ddebrunner — COMMENTED** [review]
GTG - tests pass

**Review by ddebrunner — APPROVED** [review]

**Pull request merged**